File: Data Structures & Algorithms/combination-target-sum/submission-4.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res, subset = [], []
        sum = 0

        def dfs(index, sum):
            if sum > target: return
            if sum == target:
                res.append(subset.copy())
                return
            
            for i in range(index, len(nums)):
                subset.append(nums[i])
                dfs(i, sum + nums[i])
                subset.pop()

        dfs(0, sum)
        return res

File: Data Structures & Algorithms/missing-number/submission-4.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0;
        int num_sum = 0;
        for (int i=0; i<=nums.size(); i++) {
            sum += i;
            if (i == nums.size()) break;
            num_sum += nums[i];
        }
        cout << sum << endl;
        cout << num_sum << endl;
        return sum-num_sum;
    }
};
